packages/styles/gulpfile.js

```diff
@@ -160,6 +160,48 @@ gulp.task(
   }),
 );
 
+/**
+ * Get all available components names from the components package and add them to the scss file (packages\styles\src\utilities\_not-defined.scss) which sets initial visibility to hidden (for unregistered state).
+ */
+
+gulp.task('generate-not-defined-components-scss', done => {
+  const filePath = path.join(__dirname, '../components/src/index.ts');
+  fs.readFile(filePath, 'utf8', (err, data) => {
+    if (err) {
+      done(err);
+      return;
+    }
+
+    const content = data;
+    const componentNames = content
+      .match(/export \{ (\w+) \} from/g)
+      .map(match => match.replace(/export \{ (\w+) \} from/, '$1'));
+
+    const kebabCaseNames = componentNames.map(name =>
+      name.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase(),
+    );
```

Comment:
```suggestion
    ```suggestion
    const kebabCaseNames = Array.from(data.matchAll(/export \{ (\w+) \} from/g), (m) => m[1].replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase());
```

Since you're not using the vars `data` and `componentNames` anymore later in code, you can combine everything to this single line. And also avoid implementing the regex twice.